corehq/apps/sms/resources/v0_5.py

```diff
@@ -227,3 +268,59 @@ def post_list(self, request, **kwargs):
             }),
             content_type='application/json',
         )
+
+
+class UserSelfRegistrationReinstallResource(BaseUserSelfRegistrationResource):
+    """
+    Used to resend the CommCare install link to a user's phone.
+    """
+
+    def __init__(self, *args, **kwargs):
+        super(UserSelfRegistrationReinstallResource, self).__init__(*args, **kwargs)
+        self.reinstall_result = None
+
+    class Meta:
+        authentication = RequirePermissionAuthentication(Permissions.edit_data)
+        resource_name = 'sms_user_registration_reinstall'
+        allowed_methods = ['post']
+        validation = UserSelfRegistrationReinstallValidation()
+
+    def full_hydrate(self, bundle):
+        if not self.is_valid(bundle):
+            raise ImmediateHttpResponse(response=self.error_response(bundle.request, bundle.errors))
```

Comment:
yeah that happens automatically - the `is_valid` method being called here is the method on the `Resource` superclass which ends up calling `is_valid` on the custom `Validation` class and puts the errors in `bundle.errors`